xbmc/cores/RetroPlayer/savestates/SavestateDatabase.cpp

```diff
@@ -27,20 +34,36 @@ std::unique_ptr<ISavestate> CSavestateDatabase::CreateSavestate()
   return savestate;
 }
 
-bool CSavestateDatabase::AddSavestate(const std::string& gamePath, const ISavestate& save)
+bool CSavestateDatabase::AddSavestate(std::string& savestatePath,
+                                      const std::string& gamePath,
+                                      const ISavestate& save)
 {
   bool bSuccess = false;
+  std::string path;
+
+  if (savestatePath.empty())
+  {
+    path = MakePath(gamePath);
+    path += "/";
```

Comment:
Use a path joining function instead of string concatenation. Besides the performance impact of concatting (instead you would use `stringstreams`), the `/` is not the universal path separator because windows uses `\`

Reply:
I can't find the path joining function can you help me. Btw windows translates `/` to `\`